cfme/v2v/migrations.py

```diff
@@ -0,0 +1,130 @@
+import attr
+
+from cfme.base.login import BaseLoggedInPage
+from cfme.modeling.base import BaseCollection, BaseEntity
+from cfme.utils.appliance.implementations.ui import navigator, CFMENavigateStep, navigate_to
+
+from navmazing import NavigateToAttribute, NavigateToSibling
+from widgetastic.widget import View
+from widgetastic_patternfly import Text, TextInput, Button
+from widgetastic_manageiq import InfraMappingWizard
+
+
+# class InfrastructureMappingWizard(View):
+#     initial_screen = View.nested
+# Views
+class MigrationDashboardView(BaseLoggedInPage):
+    create_infrastructure_mapping = Text(locator='(//a|//button)'
+        '[text()="Create Infrastructure Mapping"]')
+    create_migration_plan = Text(locator='(//a|//button)[text()="Create Migration Plan"]')
+
+    @property
+    def is_displayed(self):
+        return self.navigation.currently_selected == ['Compute', 'Migration']
+
+
+class AddInfrastructureMappingView(View):
+    wizard_form = InfraMappingWizard()
+
+    @property
+    def is_displayed(self):
+        return (self.wizard_form.title.text == 'Infrastructure Mapping Wizard')
+
+
+class AddMigrationPlanView(View):
+    title = Text(locator='.//h4[contains(@class,"modal-title")]')
+    name = TextInput(name='name')
+    description = TextInput(name='description')
+    back_btn = Button('Back')
+    # Since next is a keyword, suffixing it with btn and other two
+    # because want to keep it consistent
+    next_btn = Button('Next')
+    cancel_btn = Button('Cancel')
+
+    @property
+    def is_displayed(self):
+        return (self.title.text == 'Migration Plan Wizard')
+
+# Collections Entities
+
+
+@attr.s
+class InfrastructureMapping(BaseEntity):
+    """Class representing v2v infrastructure mappings"""
+    name = attr.ib()
+    description = attr.ib()
+    clusters_view = attr.ib(default=dict())
```

Comment:
It isn't clear what these are, are they view classes? Why default to dict for something that is `<name>_view` named?

Reply:
For each of these x_view=attr.ib(default=dict()), they are basically pages in the wizard/form and I pass data to fill those forms as dictionary. Do you want me to name it differently? Maybe, clusters_form_data or clusters_view_data?

Reply:
I think its a bit of an odd pattern - The attributes of InfrastructureMapping should detail items that are part of the model - specific elements of the clusters view.  Things like `source_cluster`, `target_cluster`, etc.

An `as_fill_value()` method would then return the dictionary for filling the form, built from the object's attributes.  This could be split to `clusters_fill_value`, but I think it would be better to have a single method that returns a single dict matching the form class's View+Widget structure.

This leaves navigation and creation to simple `view = navigate_to(collection, 'Add')`, `view.form.fill(infra_mapping_instance.as_fill_value())`

Reply:
the reason I did not break it down to things like source_clusters, target_clusters because it is difficult to predict those attributes upfront. What I am doing is as follows: 
 https://github.com/ManageIQ/integration_tests/pull/7236/files#diff-264354a78406ae4ca6d836bd7388d376R4269
 I need to pass in a "Mapping":{'sources':[], 'target': } and there could be more than one mappings. So best way I to depict that in the entity was to let user pass in entire dict for each view/page in the wizard, that includes all the data as follows: https://github.com/ManageIQ/integration_tests/pull/7236/files#diff-3a77e1415d5ddb90c31c2f74d6d32536R15
 and Form_data in the entity then constructs that from input here: https://github.com/ManageIQ/integration_tests/pull/7236/files#diff-3a77e1415d5ddb90c31c2f74d6d32536R15 and that can be used to fill in the values in the form
the form_data from test_v2v.py maybe removed and added to YAML as for a particular test I will need to have particular set of inputs. 
happy to discuss further. I tried to discuss this with landon and that's the best we could come up with i.e. getting data from YAML